tfjs-backend-webgpu/src/kernels/StringNGrams.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+import {KernelConfig, KernelFunc, StringNGrams, StringNGramsAttrs, StringNGramsInputs, TensorInfo} from '@tensorflow/tfjs-core';
+
+import {WebGPUBackend} from '../backend_webgpu';
+import {stringNGramsImplCPU} from '../kernel_utils/shared';
+
+export function stringNGrams(args: {
+  inputs: StringNGramsInputs,
+  backend: WebGPUBackend,
+  attrs: StringNGramsAttrs
+}): [TensorInfo, TensorInfo] {
+  const {inputs, backend, attrs} = args;
+  const {
+    separator,
+    nGramWidths,
+    leftPad,
+    rightPad,
+    padWidth,
+    preserveShortSequences
+  } = attrs;
+  const {data, dataSplits} = inputs;
+
+  if (data.dtype !== 'string') {
+    throw new Error('Data must be of datatype string');
+  }
+  if (dataSplits.dtype !== 'int32') {
+    throw new Error('Data splits must be of datatype int32');
+  }
+  if (data.shape.length !== 1) {
+    throw new Error(`Data must be a vector, saw: ${data.shape}`);
+  }
```

Comment:
L39-L47 are repeated verified on cpu/webgl/webgpu. Should this part be done in tfjs-core?

@pyu10055 @lina128 @jinjingforever What's your opinion? Also not only to this PR, there are many files. They are almost same in webgl/webgpu expect for the backendName, like this one. Do you have any suggestion to reduce the duplication?